pkg/controller/namespace/namespace_controller.go

```diff
@@ -133,7 +133,11 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// Bind to workspace if the namespace created by kubesphere
 	_, hasWorkspaceLabel := namespace.Labels[tenantv1alpha1.WorkspaceLabel]
-	if hasWorkspaceLabel {
+	// if the namespace doesn't have a label like kubefed.io/managed: "true" (single cluster environment)
+	// or it has a label like kubefed.io/managed: "false"(multi-cluster environment), we set the owner reference filed.
+	// Otherwise, we don't set the owner reference filed.
+	kubefedManaged := namespace.Labels[constants.KubefedManagedLabel] == "true"
+	if hasWorkspaceLabel && !kubefedManaged {
```

Comment:
```suggestion
	if !kubefedManaged {
		if hasWorkspaceLabel {
			if err := r.bindWorkspace(rootCtx, logger, namespace); err != nil {
				return ctrl.Result{}, err
			}
		} else {
			if err := r.unbindWorkspace(rootCtx, logger, namespace); err != nil {
				return ctrl.Result{}, err
			}
		}
	}
```
The results should be the same, but the logic seems more clear.

Reply:
fixed